main.rb

```diff
@@ -1,102 +1,146 @@
-#The Enumarable project: Creating own methods
+# The Enumarable project: Creating own methods
+
+# frozen_string_literal: true
+
+# rubocop:disable Metrics/ModuleLength
 module Enumerable
+  def my_each
+    return to_enum unless block_given?
 
-    def my_each
-      raise LocalJumpError, 'You have not given a block to the method' unless block_given?
-      for el in self
-        yield el
-      end
+    tmp = is_a?(Range) ? to_a : self
+    i = 0
+    result = []
+    while i < tmp.size
+      result << yield(tmp[i])
+      i += 1
     end
+    result
+  end
 
-    def my_each_with_index
-        raise LocalJumpError, 'You have not given a block to the method' unless block_given?
-        i = 0
-        for el in self
-            yield i,el
-            i+=1 
-        end
-    end
+  def my_each_with_index
+    return to_enum :my_each_with_index unless block_given?
 
-    def my_select
-      raise LocalJumpError, 'You have not given a block to the method' unless block_given?
-      selected_items = []
-      my_each { |i| selected_items.push(i) if yield(i) }
-      selected_items
+    result = []
+    k = 0
+    my_each do |v|
+      result << yield(k, v)
```

Comment:
```suggestion
      result << yield(v, k)
```

Let's interchange the `yield` argument. See ruby doc https://ruby-doc.org/core-2.6.5/Enumerable.html#method-i-each_with_index

> Calls block with two arguments, the item and its index...

```ruby
hash = Hash.new
%w(cat dog wombat).each_with_index { |item, index|
  hash[item] = index
}
hash   #=> {"cat"=>0, "dog"=>1, "wombat"=>2}